### PR TITLE
#79 - Inviting to quizzes frontend

### LINKED ISSUE #79: Inviting to quizzes - Frontend
Implement the frontend for the inviting page.

- [ ] Records should be selectable
- [ ] Provide an option to select all displayed records
- [ ] Provide an option to invite all users from selected records to the next stage of the contest
- [ ] Sorting records by school, score, or username
- [ ] Option to group records by school, quiz
- [ ] Option to limit the displayed records per group to a configurable number
- [ ] Pagination

### PR DESCRIPTION
**Pull request by JokeUrSelf**
Should close #79

- Added invitation to quizzes page
- Added email invitation
- Added pagination support for the homepage searchbar
- Adjusted design on a profile page
- Added theme switch to a profile page 
- Adjusted sidebar design

image
image
image
image
image
image

### CODE REVIEW

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. I'm not a big fan of this - three buttons on the same color - I believe that check/uncheck all users can be made better

image

2. As I understand I can only invite to published quiz? Is it okay for us?

3. Missing page title

image

4. For me, this icon should be on the left of the input, not on the right

image

5. When I click "Szkoły" in this search, I don't see schools, just some users :eyes: 

6. We should add email notification for students that were assigned to quiz

**Comment by JokeUrSelf** [comment]
> 5. When I click "Szkoły" in this search, I don't see schools, just some users 👀

It still searches for users, but through the school name
image


Maybe I should change option names to something like this, to make it more intuitive?
image

**Comment by AmonDeShir** [comment]
1. Sorting by schools doesn't work. It seems that there is a bug in `InviteController.php`. I think `sortBySchool` method should look like this:
```php
 private function sortBySchool(Builder $query, SortHelper $sorter): Builder
    {
        [$field, $order] = $sorter->getSortParameters();

        if ($field === "school") {
            return $query->join("schools", "users.school_id", "=", "schools.id")
                ->orderBy("schools.city", $order)
                ->orderBy("schools.name", $order)
                ->select("users.*");
        }

        return $query;
    }
```

2. I'm getting duplicate results in the school search bar. When I modify the input, the number of duplicated elements doubles.
image
image

3. If I delete the text in the search bar, I get an incorrect variant of the "no content" component. This issue occurs on every page that uses the CRUD component.
image
image
image

**Review by AmonDeShir — CHANGES_REQUESTED** [review]
1. Sorting by schools doesn't work. It seems that there is a bug in `InviteController.php`. I think `sortBySchool` method should look like this:
```php
 private function sortBySchool(Builder $query, SortHelper $sorter): Builder
    {
        [$field, $order] = $sorter->getSortParameters();

        if ($field === "school") {
            return $query->join("schools", "users.school_id", "=", "schools.id")
                ->orderBy("schools.city", $order)
                ->orderBy("schools.name", $order)
                ->select("users.*");
        }

        return $query;
    }
```

2. I'm getting duplicate results in the school search bar. When I modify the input, the number of duplicated elements doubles.
image
image

3. If I delete the text in the search bar, I get an incorrect variant of the "no content" component. This issue occurs on every page that uses the CRUD component.
image
image
image

**Comment by EwelinaSkrzypacz** [comment]
Conflicts :(

**Comment by JokeUrSelf** [comment]
> 2\. As I understand I can only invite to published quiz? Is it okay for us?

We collectively decided that it is okay for us

**Review by AmonDeShir — DISMISSED** [review]

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. When all users are selected, for me it should be here text like "Odznacz wszystkich" and when all users are not selected it should be "Zaznacz wszystkich"

image

2. Invite user page - I think I will add information somewhere to what quiz users are invited.

3. We should add email notification for students that were assigned to quiz

**Comment by AmonDeShir** [comment]
> 1. When all users are selected, for me it should be here text like "Odznacz wszystkich" and when all users are not selected it should be "Zaznacz wszystkich"

fixed

>     2. Invite user page - I think I will add information somewhere to what quiz users are invited.

fixed
 
>     3. We should add email notification for students that were assigned to quiz


fixed

File: app/Http/Controllers/InviteController.php [hunk 1]
```diff
@@ -44,11 +50,11 @@ public function assign(Quiz $quiz, InviteQuizRequest $request, AssignToQuizActio
     {
         $this->authorize("invite", $quiz);
 
-        $assignAction->execute($quiz, collect($request->input("ids")));
+        $assignAction->execute($quiz, collect($request->input("ids")), $request->user());
 
         return redirect()
             ->back()
-            ->with("status", "Użytkownicy zostali przypisani do quizu.");
+            ->with("status", "Użytkownicy zostali przypisani do testu.  Za 15 minut zostaną o tym powiadomieni drogą mailową. Jeżeli w ciągu 15 minut anulujesz zaproszenie, mail nie zostanie wysłany.");
```
Comment (EwelinaSkrzypacz) [inline app/Http/Controllers/InviteController.php:57]:
```suggestion
            ->with("status", "Użytkownicy zostali przypisani do testu. Za 15 minut zostaną o tym powiadomieni drogą mailową. Jeżeli w ciągu 15 minut anulujesz zaproszenie, e-mail nie zostanie wysłany.");
```

**Review by EwelinaSkrzypacz — COMMENTED** [review]
In e-mail, I will add some button that redirects to the quiz

image

**Review by EwelinaSkrzypacz — APPROVED** [review]

**Review by PrabuckiDominik — APPROVED** [review]

**Pull request merged**